aten/src/ATen/native/TensorShape.cpp

```diff
@@ -18,13 +18,36 @@ static void check_cat_no_zero_dim(TensorList tensors) {
   }
 }
 
+// FIXME when empty tensors with arbitrary shapes is implemented
+// When using cat on empty tensors, the following is legacy behavior
+// when the non-empty `tensor` and `empty` do not have matching sizes:
+// 1) torch.cat([empty, tensor]) should ignore empty tensors
+// 2) torch.cat([tensor, empty]) should ignore empty tensors
+// 3) torch.cat([empty, empty, ..., empty]) should be empty
+// The following function wraps dim on the first non-empty tensor
+// to preserve legacy behavior.
+static inline int64_t legacy_cat_wrap_dim(int64_t dim, TensorList tensors) {
+  if (tensors.size() == 0) {
+    return dim;
+  }
+  for (auto& tensor : tensors) {
+    if (tensor.numel() == 0) {
+      continue;
+    }
+    return at::maybe_wrap_dim(dim, tensor.dim());
+  }
+  return 0;
```

Comment:
why is this 0 instead of dim?  I guess it doesn't matter, but it feels like it should match the tensors.size() == 0 case.  And if it could matc you can get rid of the first if.

Reply:
Good point, it should be fine as `dim`. I'll try that and see how it goes